### ISSUE TITLE
Unable to get property 'contains' of undefined or null reference

### ISSUE THREAD

**Issue by esetnik** [jwplayer/jwplayer#3370]
### Expected Behavior

No error on Edge browser

### Actual Behavior

Sentry is reporting errors on Edge browser:

Unable to get property 'contains' of undefined or null reference
```
{snip}  t.hideMenu()},3e3),!1}},{key:"hideMenu",value:function(e){e&&this.el.contains(e.target)||(Object(l.n)(this.playerContainer,"jw-flag-rightcl {snip}
```

### Steps to reproduce

Unknown

### Environment

- *Windows 10*
- *Edge 17.17134*
- */player/v/8.8.2/jwplayer.core.controls.html5.js*

**Comment by jnatalzia** [comment]
Hey @esetnik do you have any additional information? I understand that Sentry is reporting errors but does this happen on all plays on this browser? Specific embeds? Certain configurations? 

It's hard for us to narrow down the issue unless we know the scope. Any additional information (especially reproduction steps) would be extremely helpful here.

**Comment by esetnik** [comment]
@jnatalzia totally understand that this is tricky to debug without steps to reproduce. I don't have any more detail about this issue right now as it's only occurred once and I can't reproduce it.

I did a bit of digging and see a potential issue which could be causing this. The offending line is most likely: https://github.com/jwplayer/jwplayer/blob/c1b9ce108832be0a7d98f19c1ea737c90aadf73e/src/js/view/controls/rightclick.js#L107
```
    hideMenu(evt) {
        if (evt && this.el.contains(evt.target)) {
            // Do not hide menu when clicking inside menu
            return;
        }

        removeClass(this.playerContainer, 'jw-flag-rightclick-open');
        removeClass(this.el, 'jw-open');
    }
```
`this.el.contains()` would only be called when `evt` is truthy. And the only time `hideMenu` is called with a value is when it is bound to `hideMenuHandler`https://github.com/jwplayer/jwplayer/blob/c1b9ce108832be0a7d98f19c1ea737c90aadf73e/src/js/view/controls/rightclick.js#L134

There are 4 instances where `hideMenuHandler` is added as an event listener:
https://github.com/jwplayer/jwplayer/blob/c1b9ce108832be0a7d98f19c1ea737c90aadf73e/src/js/view/controls/rightclick.js#L169
https://github.com/jwplayer/jwplayer/blob/c1b9ce108832be0a7d98f19c1ea737c90aadf73e/src/js/view/controls/rightclick.js#L170
https://github.com/jwplayer/jwplayer/blob/c1b9ce108832be0a7d98f19c1ea737c90aadf73e/src/js/view/controls/rightclick.js#L173
https://github.com/jwplayer/jwplayer/blob/c1b9ce108832be0a7d98f19c1ea737c90aadf73e/src/js/view/controls/rightclick.js#L174

But there are only 3 corresponding removals on `removeHideMenuHandlers`:
https://github.com/jwplayer/jwplayer/blob/c1b9ce108832be0a7d98f19c1ea737c90aadf73e/src/js/view/controls/rightclick.js#L188
https://github.com/jwplayer/jwplayer/blob/c1b9ce108832be0a7d98f19c1ea737c90aadf73e/src/js/view/controls/rightclick.js#L198
https://github.com/jwplayer/jwplayer/blob/c1b9ce108832be0a7d98f19c1ea737c90aadf73e/src/js/view/controls/rightclick.js#L199

Namely the `touchstart` event handler is never removed. So possibly this is what is causing the issue if `touchstart` event is fired after `destroy` is called?

**Comment by jnatalzia** [comment]
@esetnik given this has only occurred once and we don't know how to reproduce we can't dedicate resources on our end at this time to dig further. In the event you find reproduction steps please do reach out and we'd be more than happy to take a look! Thanks

**Comment by esetnik** [comment]
I think it's a bug that you do not remove the event listener for `touchstart` as I mentioned in my previous response. Whether or not that is the cause of this bug, it should probably be fixed as it potentially causes a dangling event listener. I'm happy to submit a PR if you like.

**Comment by jnatalzia** [comment]
When a bug like this is fixed/submitted our internal testing team needs to be able to reproduce the original issue in order to consider it complete. Without a way to reproduce we cannot confirm with confidence that we both fixed the issue and did not introduce any regressions in the process. 

Again, with only a single occurrence it is not prudent for us to put resources into determining reproduction steps on our end. If you are able to give us steps to reproduce the issue we are more than happy to take a look at that time. Thank you so much!

**Comment by esetnik** [comment]
I found a way to reproduce this consistently in my project and I was able to trace the issue using the debugger. It's a bit hard to reproduce consistently on Codesandbox because it is due to a race condition in how the browser fires a click handler. The best way this bug can be reproduced is if a click on the document body causes the player instance to be removed e.g. remove() api is called.

Steps:
1. User clicks on document body with `document.addEventListener('click', this.hideMenuHandler);`
2. This unloads the player from the dom
3. Unloading the player calls remove() which in turn calls destroy() on RightClick
4. destroy() sets this.el = null
5. Browser calls document click event handler (this.hideMenuHandler)
6. this.el is now null and hideMenu(evt) is called

Crash occurs at this.el.contains()
